documentation/error-handling.md

```diff
@@ -0,0 +1,103 @@
+# Error handling
+
+Because ui extensions script runs in non-browser environment (ex: WebWorker in Web, JavascriptCore on Mobile), it is recommended to listen and send errors to server manually. However, it is still possible to use external libraries for reporting errors. This document shows a couple of examples about how you can handle it.
+
+## The generic way of handling errors
+
+There are two types of errors when using ui extensions: lifecycle and non-lifecycle error.
+
+- **Lifecycle error:** most errors will fall into this type. It could happen when ui extensions script is loaded / imported, the script is run / rendered, or any interaction via callback of ui extensions components or apis. To catch this error, you can add event listener for `unhandledrejection`.
+
+  ```javascript
+  self.onunhandledrejection = (error) => {
+    console.warn('onunhandledrejection', error);
+  };
+  // or
+  self.addEventListener('unhandledrejection', (error) => {
+    console.warn('event unhandledrejection', error);
+  });
+  ```
+
+- **Non-lifecycle error:** this error could happen when the script throw exception while running setTimeout callback. To catch this error, you can add event listener for `error`.
+
+  ```javascript
+  self.onerror = (error) => {
+    console.warn('onerror', error);
+  };
+  // or
+  self.addEventListener('error', (error) => {
+    console.warn('event error', error);
+  });
+  ```
+
+## Handle errors via 3rd-party libraries
+
+### Bugsnag
+
+See https://docs.bugsnag.com/platforms/javascript/.
+
+You can use either CDN version or npm dependencies to run Bugsnag. However, at runtime, Bugsnag requires additional polyfill as it tries to access browser environment. Below script has been tested at the time of writing this docs
```

Comment:
```suggestion
You can run Bugsnag using the CDN version or npm dependencies. However, Bugsnag requires additional polyfills at runtime, as it will try to access the browser environment. The script below has been tested at the time of writing these docs.
```